libs/components/forms/testing/src/modules/file-attachment/file-attachment/file-attachment-harness.spec.ts

```diff
@@ -0,0 +1,544 @@
+    const { fileAttachmentHarness } = await setupTest({});
+    await expectAsync(
+      fileAttachmentHarness.clickUploadedFile(),
+    ).toBeRejectedWithError('Unable to find uploaded file.');
+  });
+
+  it('should delete the uploaded file after the delete button is clicked', async () => {
+    const { fileAttachmentHarness, fixture } = await setupTest({
+      dataSkyId: 'reactive-file-attachment',
+    });
+
+    const file = new File([], 'file.txt', { type: 'text/plain ' });
+    fixture.componentInstance.formGroup.controls['attachment'].setValue({
+      file,
+      url: 'foo.bar',
+    });
+    fixture.detectChanges();
+
+    expect(
+      fixture.componentInstance.formGroup.controls['attachment'].value,
+    ).toEqual({
+      file,
+      url: 'foo.bar',
+    });
+
+    await fileAttachmentHarness.clickUploadedFileDeleteButton();
+
+    expect(
+      fixture.componentInstance.formGroup.controls['attachment'].value,
+    ).toEqual(undefined);
+  });
+
+  it('should throw an error when attempting to click the delete button when no file is uploaded', async () => {
+    const { fileAttachmentHarness } = await setupTest({
+      dataSkyId: 'reactive-file-attachment',
+    });
+
+    await expectAsync(
+      fileAttachmentHarness.clickUploadedFileDeleteButton(),
+    ).toBeRejectedWithError(
+      "Unable to find uploaded file's delete button. Check if a file is uploaded.",
+    );
+  });
+
+  it('should get whether custom error has fired', async () => {
+    const { fileAttachmentHarness, fixture } = await setupTest({
+      dataSkyId: 'reactive-file-attachment',
+    });
+
+    fixture.detectChanges();
+
+    fixture.componentInstance.showCustomError = true;
+    fixture.componentInstance.attachment.setValue(null);
+    fixture.componentInstance.attachment.markAsTouched();
+    fixture.detectChanges();
+
+    await expectAsync(
+      fileAttachmentHarness.hasCustomError('customError'),
+    ).toBeResolvedTo(true);
+  });
+
+  it('should get whether wrong file type error has fired', async () => {
+    const { fileAttachmentHarness, fixture } = await setupTest({
+      dataSkyId: 'reactive-file-attachment',
+    });
+
+    fixture.componentInstance.acceptedTypes = 'image/png';
+
+    fixture.componentInstance.attachment.markAsTouched();
+    const file: File = new File([], 'file.jpeg', { type: 'image/jpeg' });
+    await fileAttachmentHarness.uploadFile(file);
+
+    await expectAsync(fileAttachmentHarness.hasFileTypeError()).toBeResolvedTo(
+      true,
+    );
+  });
+
+  it('should get whether max file size error has fired', async () => {
+    const { fileAttachmentHarness, fixture } = await setupTest({
+      dataSkyId: 'reactive-file-attachment',
+    });
+
+    fixture.componentInstance.maxFileSize = 30;
+    fixture.detectChanges();
+
+    const file = new File(['a'.repeat(3000)], 'file.png', {
+      type: 'image/png',
+    });
+    fixture.componentInstance.attachment.markAsTouched();
```

Comment:
ugghhh ok so unfortunately today that is not the case. the only time file attachment is marked as touched is when the button itself is clicked. i can change that so that it also happens on file drop events, but ill put a flex topic on the board for that to run it by design.

u can recreate this issue if u go to file attachment today and:
- drag a file over that has that causes that red dotted line around it to appear which shows that that file is not accepted for some reason
- and then when u drop it nothing happens. 
- but after that if u click on the button itself, the error shows up bc thats when its marked as touched.